### PR TITLE
[AutoMM] Lazy import ray

### PR DESCRIPTION
**Pull request by zhiqiangdon**
*Issue #, if available:*
#2903 

*Description of changes:*
Lazy import ray.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by sxjscience — APPROVED** [review]
LGTM

File: multimodal/src/autogluon/multimodal/presets.py [hunk 1]
```diff
@@ -1,7 +1,5 @@
 from typing import List, Optional
 
-from ray import tune
```
Comment (Innixma) [inline multimodal/src/autogluon/multimodal/presets.py:3]: Did you test uninstalling ray and running AutoMM? This should work so long as you aren't using HPO.

File: multimodal/src/autogluon/multimodal/presets.py [hunk 2]
```diff
@@ -74,11 +78,14 @@ def default(presets: str = DEFAULT):
 
     presets, use_hpo = parse_presets_str(presets)
     if use_hpo:
+        default_tunable_hyperparameters, default_hyperparameter_tune_kwargs = get_default_hpo_setup()
         hyperparameters.update(default_tunable_hyperparameters)
         hyperparameter_tune_kwargs.update(default_hyperparameter_tune_kwargs)
 
     if presets in [HIGH_QUALITY, DEFAULT]:
         if use_hpo:
+            from ray import tune
```
Comment (Innixma) [inline multimodal/src/autogluon/multimodal/presets.py:87]: consider using `try_import_ray` for a more informative import error message.

```
from autogluon.core.utils.try_import import try_import_ray  # top of file


def import_ray_tune():
    try_import_ray()
    from ray import tune
    return tune


def my_func():
    tune = import_ray_tune()
    ...

```

**Review by Innixma — APPROVED** [review]
LGTM, but added a few comments

File: multimodal/src/autogluon/multimodal/presets.py (hunk 1 above)
Comment (zhiqiangdon) [reply multimodal/src/autogluon/multimodal/presets.py:3]: Tested. It works without doing HPO.

File: multimodal/src/autogluon/multimodal/presets.py (hunk 2 above)
Comment (zhiqiangdon) [reply multimodal/src/autogluon/multimodal/presets.py:87]: If using `try_import_ray ()`, the following error message shows.

`ImportError: ray is required to train folds in parallel. A quick tip is to install via `pip install ray==2.2.0`, or use sequential fold fitting by passing `sequential_local` to `ag_args_ensemble` when calling tabular.fitFor example: `predictor.fit(..., ag_args_ensemble={'fold_fitting_strategy': 'sequential_local'})``

which is more customized for tabular predictor and may bring confusions for AutoMM users.

Comment (Innixma) [reply multimodal/src/autogluon/multimodal/presets.py:87]: good point, then it is ok as is

**Comment by Innixma** [comment]
LGTM, feel free to merge whenever ready

**Pull request merged**